# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalChild.cpp
# Commit: d07def1365a4
# Full Hash: d07def1365a49dc8ecbf5b6c10fdfbd5ec750932
# Author: pstanciu <pstanciu@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing WPT failures @lfedcm-identity.create-store-collect.tentative.sub.https.html
#   
#   This reverts commit 03d4474e360f28be0fb075379f4e4535bd34b252.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4e25ddaadc86 -r d07def1365a4 dom/ipc/WindowGlobalChild.cpp
--- a/dom/ipc/WindowGlobalChild.cpp	Wed Jun 11 13:42:13 2025 +0000
+++ b/dom/ipc/WindowGlobalChild.cpp	Wed Jun 11 18:04:27 2025 +0300
@@ -575,6 +575,26 @@
   return IPC_OK();
 }
 
+IPCResult WindowGlobalChild::RecvNavigateForIdentityCredentialDiscovery(
+    const nsCString& aURI, const IdentityLoginTargetType& aType) {
+  AutoJSAPI jsapi;
+  if (!jsapi.Init(GetWindowGlobal())) {
+    return IPC_OK();
+  }
+  MOZ_ASSERT(WindowContext()->TopWindowContext());
+  nsGlobalWindowOuter* outer = nsGlobalWindowOuter::GetOuterWindowWithId(
+      WindowContext()->TopWindowContext()->OuterWindowId());
+  bool popup = aType == IdentityLoginTargetType::Popup;
+  RefPtr<dom::BrowsingContext> newBC;
+  if (popup) {
+    Unused << outer->OpenJS(aURI, u"_blank"_ns, u"popup"_ns,
+                            getter_AddRefs(newBC));
+    return IPC_OK();
+  }
+  Unused << outer->OpenJS(aURI, u"_top"_ns, u""_ns, getter_AddRefs(newBC));
+  return IPC_OK();
+}
+
 void WindowGlobalChild::SetDocumentURI(nsIURI* aDocumentURI) {
   // Registers a DOM Window with the profiler. It re-registers the same Inner
   // Window ID with different URIs because when a Browsing context is first